### PR TITLE
[fleet_server] Make package compatible with 8.0

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

File: packages/fleet_server/manifest.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+version: 1.1.0
 release: ga
 description: Centrally manage Elastic Agents with the Fleet Server integration
 type: integration
 format_version: 1.0.0
 license: basic
 categories: ["elastic_stack"]
 conditions:
-  kibana.version: "^7.14.0"
+  kibana.version: "^7.16.0 || ~8.0.0"
```
Comment (ruflin) [inline packages/fleet_server/manifest.yml:11]: @jlind23 As soon as 8.0 lands, we need to follow up with a package for 8.1.0 or decide how far in the future we want to be compatible.

Comment (jlind23) [reply packages/fleet_server/manifest.yml:11]: Why did you fix version to 8.0.0 and not something like 8.X ?

Comment (ruflin) [reply packages/fleet_server/manifest.yml:11]: This gives us time to figure out the long term compatiblity story (see https://github.com/elastic/elastic-package/pull/571)

**Review by mtojek — APPROVED** [review]
LGTM.

Please remember to promote the package to the production stage.

**Comment by ruflin** [comment]
I updated this PR based on my comment in https://github.com/elastic/elastic-package/pull/571#issuecomment-961750162 Likely needs another round of review.

**Review by jsoriano — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]

**Comment by ruflin** [comment]
Getting this in and promoting to staging. I'll push to production as soon as @andresrc also gives his 👍

**Pull request merged**